app/controllers/concerns/course/discussion/posts_concern.rb

```diff
@@ -30,6 +38,17 @@ def destroy
   # Render a new post in a separate page
   # This can be used when you do not want to introduce javascript to your implementation
   def reply
+    @reply_post = discussion_topic.posts.build(title: t('course.discussion.posts.reply_title',
+                                                        title: @post.title))
+  end
+
+  # The same function as create. But it will render 'reply' partial when creating fails
+  def create_reply
```

Comment:
I'm a bit concerned about adding this method, if it's only for `render 'reply'`. 

And do we need to handle subscriptions here ?